test/conformance/source/DevicesEnvironmentMain.cpp

```diff
@@ -0,0 +1,11 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#include <uur/environment.h>
+
+int main(int argc, char **argv) {
+  auto *environment = new uur::DevicesEnvironment(argc, argv);
```

Comment:
I think we should just always have our own `main.cpp` and use a macro defined in CMake to switch between which environment class to instantiate. This way we don't have mostly identical code and two different conditinal logic in CMake to sometimes link `gtest_main`.

Reply:
Would you like individual an `main.cpp` for each test suite, or a single global main shared by every testsuite? Also by CMake macro do you mean define a compiler definition with CMake or a `macro()`?

My thought was something like the following:
```cpp
#include <uur/environment.h>

int main(int argc, char **argv) {
#ifdef DEVICES_ENVIRONMENT
  auto *environment = new uur::DevicesEnvironment(argc, argv);
#endif
#ifdef PLATFORM_ENVIRONMENT
  auto *environment = new uur::PlatformEnvironment(argc, argv);
#endif
  ::testing::InitGoogleTest(&argc, argv);
#if defined(DEVICES_ENVIRONMENT) || defined(PLATFORM_ENVIRONMENT)
  ::testing::AddGlobalTestEnvironment(environment);
#endif
  return RUN_ALL_TESTS();
}
```

Reply:
A shared `main.cpp` makes senese for now. If we need separate ones later we can modify things. Yes, `#ifdef`s in the code is what I was aiming for.